master/internal/db/postgres_tasks.go

```diff
@@ -189,10 +194,28 @@ func (db *PgDB) UpdateAllocationState(a model.Allocation) error {
 	return err
 }
 
+// UpdateAllocationStartTime stores the latest start time.
+func (db *PgDB) UpdateAllocationStartTime(a model.Allocation) error {
+	_, err := db.sql.Exec(`
+		UPDATE allocations
+		SET state=$2
+		WHERE allocation_id=$1
+	`, a.AllocationID, a.StartTime)
+	return err
+}
+
 // CloseOpenAllocations finds all allocations that were open when the master crashed
 // and adds an end time.
 func (db *PgDB) CloseOpenAllocations() error {
 	if _, err := db.sql.Exec(`
+	UPDATE allocations
```

Comment:
nit, here and elsewhere format the queries consistently. also i would run these queries in a transaction.

Reply:
will do!

Reply:
locally, ran the queries in a transaction and they work fine!